### DISCUSSION TITLE
Kernel panic and ubuntu so-setup errors

### DISCUSSION THREAD

**Discussion by kiot389** [Security-Onion-Solutions/securityonion#6733] [Unsupported Versions] [answered]
I can't identify why SO is failing to install two different systems. I get a kernel panic if I use the image, and if I install CentOS or Ubuntu so-setup fails. I just built a system for SO; AMD Ryzen 5 5800x, ASUS B550-A, 16GB DDR4, Intel 512 M.2. The other is a Dell OptiPlex. There's no proxy or firewalls to the Internet. Is someone able to review the sosetup.log I attached?

**Comment by dougburks** [comment]
> Is someone able to review the sosetup.log I attached?

I'm not seeing any attachments.

**Comment by kiot389** [comment]
[sosetup.log](https://github.com/Security-Onion-Solutions/securityonion/files/7802756/sosetup.log)

**Reply by kiot389** [reply]
@dougburks My mistake, bad upload, should be there now.

**Comment by dougburks** [comment]
Our ISO image is the recommended installation method:
https://docs.securityonion.net/en/2.3/download.html

Can you provide more information about the kernel panic that you're seeing with the ISO?

Did you verify the downloaded ISO?

Did you use Balena Etcher to create a USB?

If you still can't get our ISO image to work, please try CentOS 7 and provide its sosetup.log.

**Comment by kiot389** [comment]
The hash is good and imaged with Balena. Attached is the boot screen. I can try to get a full kernel log with something like kdump, but it'll be tricky
![kernl boot screen](https://user-images.githubusercontent.com/36932098/148437546-74f71e6f-d160-441a-a305-a36783b55803.jpg)
.

**Reply by kiot389** [reply]
The issue is AMD's CentOS support. This is the first time I ever used AMD and can't believe how many issues its caused. CentOS forum user had the same platform issue: " just got B550 + 5900X today, can't install 7 on it." Further 2017 discussion is: "if Ryzen 5 hasn't supported CentOS 7 yet, they probably never will." 

I appreciate you looking into this @dougburks Who would have thought AMD Ryzan 5 wouldn't run CentOS 7? Regardless 7.1-7.3 might work, if I get it installed, I'll update, and drop a sosetup.log, but it's not necessary to check out. CentOS 8 runs fine. If anyone has this Kernel panic and is running AMD...unbelievable. Will keep an eye out for SO's migration from CentOS. How are you gonna manage Security Onion since CentOS is retired?

**Reply by acts1417** [reply]
thanks @kiot389  . I just ran into the same for the ISO a week ago.  Right now i'm trying the sosetup with ubuntu but the log ends with: 
the following packages will be DOWNGRADED:
  docker-ce docker-ce-cli docker-ce-rootless-extras
0 upgraded, 0 newly installed, 3 downgraded, 0 to remove and 0 not upgraded. (100)
Command failed with exit code 100; will retry in 10 seconds (50 / 50)...
Command continues to fail; giving up.
Errors detected during setup; skipping post-setup steps to allow for analysis of failures.

**Reply by acts1417** [reply]
i seemed to have gotten farther by first 'sudo apt remove docker-ce docker-ce-cli docker-ce-rootless-extras'  instead of having the script try to downgrade them.  
The install script still ended with: [snip]
Summary for local
--------------
Succeeded: 610 (changed=63)
Failed:      0
--------------
Total states run:     610
Total run time:   343.969 s
Errors detected during setup; skipping post-setup steps to allow for analysis of
 failures.
=========================================================================
Stopping mysql...
=========================================================================
so-mysql
so-mysql

[/snip]

The only warnings were things like: 
[WARNING ] State for file: /etc/pki/ca.crt - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
There was a : curl: (7) Failed to connect to localhost port 3000: Connection refused'} but Grafana looks fine.  

I didn't see anything in errors.log Grepping for "Failed" didn't show non-zero values.  
Rebooting my ubuntu, everything seems to check out.  
There's a popup warning from ubuntu that docker-ce "is not an official Ubuntu package" but it doesn't look to be a show-stopper atm.

**Reply by kiot389** [reply]
@acts1417 I'm installing on Ubuntu now. I got to a similar spot, but got past it by entering --allow-downgrades in a different terminal.
"apt-get -y install --allow-downgrades docker-ce=5:20...."

Command failed with exit code 100; will retry in 10 seconds (10 / 50)...
Executing command with retry support: apt-get -y install docker-ce=5:20.10.5~3-0~ubuntu-bionic docker-ce-cli=5:20.10.5~3-0~ubuntu-bionic docker-ce-rootless-extras=5:20.10.5~3-0~ubuntu-bionic python3-docker
E: Packages were downgraded and -y was used without --allow-downgrades.

**Comment by kiot389** [comment]
@acts1417 Did Ubuntu stop at any other points? Mine stops at 64% about elasticsearch.

**Comment by kiot389** [comment, marked as answer]
The issue is AMD's CentOS support. CentOS forum states Ryzen 5 5800x isn't supported by CentOS 7. 

I'm doing a network based install on Ubuntu. I've ran into a couple issues, but have been able to troubleshoot and almost fix all of them.
Nearly there. Problems have been running --allow-downgrades, elasticsearch search not being installed by the script. I installed it myself, and got the install to 96%. Idk what state I should put elastic in when I run setup though, so it can still fail. Ubuntu 18 setup worked with this hardward.
